src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
+            {
+                try
+                {
+                    object result = CachedExpressionCompiler.Process(expression)((TModel)container);
+                    if (expression.ReturnType.GetTypeInfo().IsEnum)
+                    {
+                        var enumValue = metadata.EnumNamesAndValues[result.ToString()];
```

Comment:
Won't `enumValue == result` be `true` in all cases except when the `enum` has a value that doesn't match one of its fields e.g. `(StringComparison)23`?